java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java

```diff
@@ -65,23 +66,45 @@ public void visitNode(Tree tree) {
   }
 
   private static ModifierTree getFirstBadlyOrdered(ModifiersTree modifiersTree) {
-    int modifierIndex = -1;
+    ListIterator<ModifierTree> modifiersIterator = modifiersTree.listIterator();
+    skipAnnotations(modifiersIterator);
```

Comment:
As discussed, this is in fact equivalent and having it in the same place allows to simplify previous logic a bit.